### DISCUSSION TITLE
Announcing ox-neuron, and thank you for Neuron

### DISCUSSION THREAD

**Discussion by vedang** [srid/neuron#644] [General]
Hello @srid ,

I am an avid Org-Mode user and I use a package called `org-brain` (https://github.com/Kungsgeten/org-brain/) for taking Zettelkasten notes. It works wonderfully for my private notes and local note management, but it lacks a killer static site generator. For my actual site, I use `ox-hugo` (https://github.com/kaushalmodi/ox-hugo), which is a great tool but doesn't support the Zettelkasten method. 

So I was overjoyed when I came across Neuron a few weeks ago. It was the perfect solution : a gorgeous website for visualizing Zettelkasten notes. I also happened to have the perfect project for publication - class notes for a very detailed and rigorous class I was part of. 

I quickly hacked up an exporter over the weekends for my notes. This is `ox-neuron` and is available at https://github.com/vedang/ox-neuron. It works beautifully for my needs and currently exports Org's ID, Custom_ID and org-brain's brain-parent, brain-child and brain-friend relationships perfectly. Extending it to handle `Org-Roam` links should be straightforward as well, but I'll leave that to some Roam user.

I am currently in the process of moving my personal site from `ox-hugo` to `ox-neuron` and the class notes are currently private (pending approval for publication). I'll share links to both sites as and when I can.

Thank you for building this tool! I see that it is now in maintenance mode in favor of Emanote and I'm sad about this, but based on what I read `ox-neuron` should work just fine with Emanote eventually. (Or I'll build a new exporter once Emanote is ready)

**Comment by vedang** [comment]
I have one minor feedback to share based on my experience with writing the exporter so far. I am using the `dirtree` plugin because for me, exporting notes in the Local Parent style just makes sense (i.e heading contents are exported to a file, and then all subheadings are exported to `directories` based on their nesting under the original heading). This establishes parent children relationships is a very elegant manner imho. However, since I convert Org Mode IDs to Neuron IDs (filenames as well as directory names), this approach leads to each entry having ugly tags (example: `root/0fd23451` and `root/0fd23451/2wert345` etc).

To avoid having these tags on the static site, I have turned off the tags plugin at the moment. Ideally, I'd like an option to control whether the dirtree plugin should generate directory path based tags or not, so that I can turn them off altogether for dirtree and still use them everywhere else.

**Reply by srid** [reply]
Great, I've added it to https://github.com/srid/neuron/commit/998fce27ccc91231ef9757e2bebeb39327850092

As for Emanote, your ox-neuron should just work unless you are using `z:zettels` queries. See, https://note.ema.srid.ca/guide/neuron

**Comment by vedang** [comment]
Looking at the issues and PRs now (something I probably should have done earlier), I see that there is some discussion around org-mode in #557 . Adding support via exporting from Org to Neuron is a different way to think about the same problem, so linking it here for anyone interested.
